message/response.go

```diff
@@ -3,34 +3,24 @@ package message
 import "encoding/json"
 
 type ErrorResponse struct {
-	Code        int
-	Description string
+	Code        int    `json:"code"`
+	Description string `json:"description"`
 }
 
-type ResponseWithGenResult struct {
-	Jsonrpc string
-	Result  int
-	Id      int
-}
-type ResponseWithCatchupResult struct {
-	Jsonrpc string
-	Result  string
-	Id      int
-}
-type ResponseWithError struct {
-	Jsonrpc       string
-	ErrorResponse json.RawMessage
-	Id            int
-}
+//We still need ResponseIDNotDecoded and ResponseWithGenResult because we cannot omit integers
 type ResponseIDNotDecoded struct {
-	Jsonrpc       string
-	ErrorResponse json.RawMessage
-	Id            []byte
+	Jsonrpc       string          `json:"jsonrpc"`
+	ErrorResponse json.RawMessage `json:"result,omitempty"`
```

Comment:
this field has the same name as a struct... --> unclear code, changing it's name would be better (same for line 23)

Reply:
@wouewouel tu as une suggestion ? pourquoi on l'appelle pas juste "Result" d'ailleurs ?

Reply:
c'est renommé, btw ça devait pas être `json:"result,omitempty"` mais bien "error" donc t'as trouvé une coquille :)

Reply:
nice